packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-cmk-with-cwl-encryption.ts

```diff
@@ -0,0 +1,97 @@
+    this.logGroupKey = new kms.Key(this, 'LogGroup Key');
+
+    this.logGroup = new logs.LogGroup(this, 'MyLogGroup', {
+      logGroupName: '/aws/vendedlogs/states/MyLogGroup',
+      encryptionKey: this.logGroupKey,
+    });
+
+    this.logGroupKey.addToResourcePolicy(new cdk.aws_iam.PolicyStatement({
+      resources: ['*'],
+      actions: ['kms:Encrypt*', 'kms:Decrypt*', 'kms:ReEncrypt*', 'kms:GenerateDataKey*', 'kms:Describe*'],
+      principals: [new cdk.aws_iam.ServicePrincipal(`logs.${cdk.Stack.of(this).region}.amazonaws.com`)],
+      conditions: {
+        ArnEquals: {
+          'kms:EncryptionContext:aws:logs:arn': cdk.Stack.of(this).formatArn({
+            service: 'logs',
+            resource: 'log-group',
+            sep: ':',
+            resourceName: '/aws/vendedlogs/states/MyLogGroup', // Cannot use this.logGroup.logGroupName since this will cause a circular dependency
+          }),
+        },
+      },
+    }));
+
+    this.stateMachine = new sfn.StateMachine(this, 'StateMachineWithCMKWithCWLEncryption', {
+      stateMachineName: 'StateMachineWithCMKWithCWLEncryption',
+      definitionBody: sfn.DefinitionBody.fromChainable(sfn.Chain.start(new sfn.Pass(this, 'PassState'))),
+      stateMachineType: sfn.StateMachineType.STANDARD,
+      kmsKey: this.kmsKey,
+      kmsDataKeyReusePeriodSeconds: cdk.Duration.seconds(300),
+      logs: {
+        destination: this.logGroup,
+        level: sfn.LogLevel.ALL,
+        includeExecutionData: false,
+      },
+    });
+  }
+}
+const app = new cdk.App();
+const stack = new KMSStateMachine(app, 'aws-stepfunctions-statemachine-and-activity-with-cmk-encryptionconfig');
+
+const testCase = new IntegTest(app, 'StateMachineAndActivityWithCMKEncryptionConfiguration', {
+  testCases: [stack],
+});
+
+// Start an execution
+const start = testCase.assertions.awsApiCall('StepFunctions', 'startExecution', {
+  stateMachineArn: stack.stateMachine.stateMachineArn,
+});
+// describe the results of the execution
+const describe = testCase.assertions.awsApiCall('StepFunctions', 'describeExecution', {
+  executionArn: start.getAttString('executionArn'),
+  includedData: 'METADATA_ONLY',
+});
+
+start.next(describe);
+
+// assert the results
+describe.expect(ExpectedResult.objectLike({
+  status: 'SUCCEEDED',
+})).waitForAssertions({
+  interval: cdk.Duration.seconds(10),
+  totalTimeout: cdk.Duration.minutes(5),
+});
+
+const logStreamNamePrefix = 'states/StateMachineWithCMKWithCWLEncryption';
+
+const describeLogStream = testCase.assertions.awsApiCall('CloudWatchLogs', 'describeLogStreams', {
+  logGroupName: stack.logGroup.logGroupName,
+  logStreamNamePrefix: logStreamNamePrefix,
+});
+
+describeLogStream.provider.addToRolePolicy({
+  Effect: 'Allow',
+  Action: ['kms:Decrypt'],
+  Resource: [stack.logGroupKey.keyArn],
+});
+
+describe.next(describeLogStream);
+describeLogStream.assertAtPath('logStreams.0.logStreamName', ExpectedResult.stringLikeRegexp(logStreamNamePrefix));
```

Comment:
Checking that a log stream exists with the state machine name doesn't necessarily tell us there's any events there.

Reply:
Discussed with wong-a offline that we'll use the `filterLogEvents` to get the events and assert the events contain a specific output from the statemachine execution.